clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -915,25 +915,25 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
 
   ModulePassManager MPM;
 
-  // FIXME: Change this when -fno-sycl-early-optimizations is not tied to
-  // -disable-llvm-passes.
-  if (CodeGenOpts.DisableLLVMPasses && LangOpts.SYCLIsDevice)
+  if (LangOpts.SYCLIsDevice && !CodeGenOpts.DisableSYCLEarlyOpts)
     MPM.addPass(SYCLPropagateAspectsUsagePass());
 
   if (!CodeGenOpts.DisableLLVMPasses) {
+
     // Map our optimization levels into one of the distinct levels used to
     // configure the pipeline.
     OptimizationLevel Level = mapToLevel(CodeGenOpts);
 
-    if (LangOpts.SYCLIsDevice)
+    if (LangOpts.SYCLIsDevice && !CodeGenOpts.DisableSYCLEarlyOpts)
```

Comment:
Those passes should also be launched even if early optimizations are disabled, because they are functional passes. We should add a test which checks that even if `-fno-sycl-early-optimizations` are present, those passes are still scheduled